# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/DeleteRangeTransaction.cpp
# Commit: 59b89f27d204
# Full Hash: 59b89f27d20494082b7a3b5c313166eefd0d3dd7
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2022-12-15 09:27:59
# Description:
#   Bug 1803336: Simplified check if a range to delete is start/end of a text node. r=masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164360
# ==============================================================================

diff -r 7ff758e0d08b -r 59b89f27d204 editor/libeditor/DeleteRangeTransaction.cpp
--- a/editor/libeditor/DeleteRangeTransaction.cpp	Wed Dec 14 07:56:22 2022 +0000
+++ b/editor/libeditor/DeleteRangeTransaction.cpp	Wed Dec 14 08:21:29 2022 +0000
@@ -57,11 +57,10 @@
   EditorRawDOMPoint startPoint(aRange.StartRef());
   EditorRawDOMPoint endPoint(aRange.EndRef());
   const bool maybeRangeStartsAfterWhiteSpace =
-      startPoint.IsInTextNode() &&
-      !(startPoint.IsStartOfContainer() && !startPoint.IsEndOfContainer());
+      startPoint.IsInTextNode() && !startPoint.IsStartOfContainer();
   const bool maybeRangeEndsAtWhiteSpace =
-      endPoint.IsInTextNode() &&
-      !(endPoint.IsEndOfContainer() && !endPoint.IsStartOfContainer());
+      endPoint.IsInTextNode() && !endPoint.IsEndOfContainer();
+
   if (!maybeRangeStartsAfterWhiteSpace && !maybeRangeEndsAtWhiteSpace) {
     // no whitespace before or after word => nothing to do here.
     return NS_OK;
